pkg/tfshim/walk/walk.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package walk
+
+import (
+	"fmt"
+	"strings"
+
+	hcty "github.com/hashicorp/go-cty/cty"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/schema"
+)
+
+// Represents locations in a tfshim.Schema value as a sequence of steps to locate it.
+//
+// An empty SchemaPath represents the current location.
+//
+// Values of this type are immutable by convention, use Copy as necessary for local mutations.
+type SchemaPath []SchemaPathStep
+
+func (p SchemaPath) GoString() string {
+	parts := []string{"walk", "NewSchemaPath()"}
+	for _, e := range p {
+		switch ee := e.(type) {
```

Comment:
Each `case` body is a new scope, so you can just shadow `e`.
```suggestion
		switch e := e.(type) {
```